File: backend/OrderManagement.Api/Models/Order.cs
namespace OrderManagement.Api.Models;

public class Order
{
    public int Id { get; set; }

    public int CustomerId { get; set; }

    public Customer Customer { get; set; } = null!;

    public OrderStatus Status { get; set; } = OrderStatus.Naujas;

    public decimal TotalAmount { get; set; }

    public DateTime CreatedAtUtc { get; set; }

    public ICollection<OrderItem> OrderItems { get; set; } = new List<OrderItem>();
}

File: frontend/src/types/Order.ts
export const ORDER_STATUSES = [
  'Naujas',
  'Vykdomas',
  'Įvykdytas',
  'Atšauktas',
] as const

export type OrderStatus = (typeof ORDER_STATUSES)[number]

export type OrderItem = {
  productId: number
  productName: string
  quantity: number
  unitPrice: number
  lineTotal: number
}

export type Order = {
  id: number
  customerId: number
  customerName: string
  status: OrderStatus
  totalAmount: number
  createdAtUtc: string
  items: OrderItem[]
}

export type CreateOrderPayload = {
  customerId: number
  items: Array<{
    productId: number
    quantity: number
  }>
}
